src/ActivitiesSDKAdapter.js

```diff
@@ -169,29 +179,51 @@ export default class ActivitiesSDKAdapter extends ActivitiesAdapter {
    * @returns {Observable.<Activity>} Observable that emits the posted activity (including id)
    */
   postActivity(activity) {
-    logger.debug('ACTIVITY', undefined, 'postActivity()', ['called with', {activity}]);
+    let activity$;
     const card = this.getAdaptiveCard(activity);
 
-    const object = card && {
-      cards: [JSON.stringify(card)],
-      displayName: activity.text,
-    };
+    if (card) {
+      activity$ = from(this.fetchConversation(activity.roomID)).pipe(
+        concatMap(async (conversation) => {
+          const encryptedCard = await this.datasource.internal.encryption.encryptText(
+            conversation.defaultActivityEncryptionKeyUrl,
+            JSON.stringify(card),
+          );
+          const object = {
+            cards: [encryptedCard],
+            displayName: activity.text,
+          };
 
-    const {id, cluster = 'us'} = deconstructHydraId(activity.roomID);
+          const sdkActivityPromiseValue = await this.datasource.internal.conversation.post(
```

Comment:
this code block (conversation.post()) repeats in the else block. We should factor it out.